activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -74,6 +74,12 @@ def options
 
         private
 
+        def id_to_index_map(ids)
```

Comment:
Should this be pushed into `ActiveRecord::Associations::Preloader::ThroughAssociation` since it's only used there?